app/js/trouble.js

```diff
@@ -53,15 +61,16 @@
 			$dropdown.data("status", 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
 			Array.prototype.foreach = function(cb) {
+        var i;
 				for (i = 0; i < this.length; ++i) {
 					cb(this[i]);
 				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			};
+			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
This was a tricky one for just about everybody. While this does remove the "making" of a function in a loop, in this case the desired change is to either A. remove the anonymous function from the loop entirely, either declaring a new function or assigning the anonymous function expression to a variable, then passing in the function or variable name as the first argument to setTimeout or B. just removing the infinite while loop. By invoking the function here, it will run as soon as the code executes rather than when the timeout event fires later.